[PATCH] cilk: fix build without wchar

When building against uClibc with wchar support disabled, WCHAR_MIN and
WCHAR_MAX are not defined leading to compilation errors.

Fix it by only including the wchar code if available.

Signed-off-by: Peter Korsgaard <peter@korsgaard.com>
---
 libcilkrts/include/cilk/reducer_min_max.h |    8 ++++++++
 1 file changed, 8 insertions(+)

Index: host-gcc-final-4.9.2/libcilkrts/include/cilk/reducer_min_max.h
===================================================================
--- host-gcc-final-4.9.2.orig/libcilkrts/include/cilk/reducer_min_max.h
+++ host-gcc-final-4.9.2/libcilkrts/include/cilk/reducer_min_max.h
@@ -3154,7 +3154,9 @@
 CILK_C_REDUCER_MAX_INSTANCE(char,               char,       CHAR_MIN)
 CILK_C_REDUCER_MAX_INSTANCE(unsigned char,      uchar,      0)
 CILK_C_REDUCER_MAX_INSTANCE(signed char,        schar,      SCHAR_MIN)
+#ifdef WCHAR_MIN
 CILK_C_REDUCER_MAX_INSTANCE(wchar_t,            wchar_t,    WCHAR_MIN)
+#endif
 CILK_C_REDUCER_MAX_INSTANCE(short,              short,      SHRT_MIN)
 CILK_C_REDUCER_MAX_INSTANCE(unsigned short,     ushort,     0)
 CILK_C_REDUCER_MAX_INSTANCE(int,                int,        INT_MIN)
@@ -3306,7 +3308,9 @@
 CILK_C_REDUCER_MAX_INDEX_INSTANCE(char,               char,       CHAR_MIN)
 CILK_C_REDUCER_MAX_INDEX_INSTANCE(unsigned char,      uchar,      0)
 CILK_C_REDUCER_MAX_INDEX_INSTANCE(signed char,        schar,      SCHAR_MIN)
+#ifdef WCHAR_MIN
 CILK_C_REDUCER_MAX_INDEX_INSTANCE(wchar_t,            wchar_t,    WCHAR_MIN)
+#endif
 CILK_C_REDUCER_MAX_INDEX_INSTANCE(short,              short,      SHRT_MIN)
 CILK_C_REDUCER_MAX_INDEX_INSTANCE(unsigned short,     ushort,     0)
 CILK_C_REDUCER_MAX_INDEX_INSTANCE(int,                int,        INT_MIN)
@@ -3432,7 +3436,9 @@
 CILK_C_REDUCER_MIN_INSTANCE(char,               char,       CHAR_MAX)
 CILK_C_REDUCER_MIN_INSTANCE(unsigned char,      uchar,      CHAR_MAX)
 CILK_C_REDUCER_MIN_INSTANCE(signed char,        schar,      SCHAR_MAX)
+#ifdef WCHAR_MAX
 CILK_C_REDUCER_MIN_INSTANCE(wchar_t,            wchar_t,    WCHAR_MAX)
+#endif
 CILK_C_REDUCER_MIN_INSTANCE(short,              short,      SHRT_MAX)
 CILK_C_REDUCER_MIN_INSTANCE(unsigned short,     ushort,     USHRT_MAX)
 CILK_C_REDUCER_MIN_INSTANCE(int,                int,        INT_MAX)
@@ -3584,7 +3590,9 @@
 CILK_C_REDUCER_MIN_INDEX_INSTANCE(char,               char,       CHAR_MAX)
 CILK_C_REDUCER_MIN_INDEX_INSTANCE(unsigned char,      uchar,      CHAR_MAX)
 CILK_C_REDUCER_MIN_INDEX_INSTANCE(signed char,        schar,      SCHAR_MAX)
+#ifdef WCHAR_MAX
 CILK_C_REDUCER_MIN_INDEX_INSTANCE(wchar_t,            wchar_t,    WCHAR_MAX)
+#endif
 CILK_C_REDUCER_MIN_INDEX_INSTANCE(short,              short,      SHRT_MAX)
 CILK_C_REDUCER_MIN_INDEX_INSTANCE(unsigned short,     ushort,     USHRT_MAX)
 CILK_C_REDUCER_MIN_INDEX_INSTANCE(int,                int,        INT_MAX)
